src/v3/src/hooks/useOnSubmit.ts

```diff
@@ -197,6 +199,16 @@ export const useOnSubmit = (): (options: OnSubmitHandlerOptions) => Promise<void
     if (step === IDX_STEP.CANCEL_TRANSACTION) {
       SessionStorage.removeStateHandle();
     }
+    if (step === IDX_STEP.IDENTIFY && features?.deviceFingerprinting) {
+      const baseUrl = getBaseUrl(widgetProps);
+      if (baseUrl) {
+        // Proceeds with form submission even if device fingerprinting fails
+        const fingerprint = await generateDeviceFingerprint(baseUrl);
```

Comment:
Do you need to catch errors here and return an empty string (or undefined) from the catch block? I am thinking if the function rejects and throws an error, this would halt the remaining progress of this hook which I don't think should happen.

Reply:
something like: 
```suggestion
        const fingerprint = await generateDeviceFingerprint(baseUrl).catch(() => undefined);
```

Reply:
Oh yeah, didn't port this over from the gen2 code, thanks